core/trino-main/src/main/java/io/trino/operator/TableFunctionOperator.java

```diff
@@ -275,23 +291,42 @@ public TableFunctionOperator(
         this.processEmptyInput = !pruneWhenEmpty;
 
         PagesIndex pagesIndex = pagesIndexFactory.newPagesIndex(sourceTypes, expectedPositions);
-        HashStrategies hashStrategies = new HashStrategies(pagesIndex, partitionChannels, prePartitionedChannels, sortChannels, sortOrders, preSortedPrefix);
-
-        this.outputPages = pageBuffer.pages()
-                .transform(new PartitionAndSort(pagesIndex, hashStrategies, processEmptyInput))
-                .flatMap(groupPagesIndex -> pagesIndexToTableFunctionPartitions(
-                        groupPagesIndex,
-                        hashStrategies,
-                        tableFunctionProvider,
-                        session,
-                        functionHandle,
-                        properChannelsCount,
-                        passThroughSourcesCount,
-                        requiredChannels,
-                        markerChannels,
-                        passThroughSpecifications,
-                        processEmptyInput))
-                .flatMap(TableFunctionPartition::toOutputPages);
+        if (partitionChannels.isEmpty()) {
```

Comment:
Do we want to to check if the `List` is empty as well? I tested locally on some custom table functions by cherry-picking this commit and ran into this case.

Reply:
my understanding is that empty list is legitimate, and requires global aggregation (one partition)